### PR TITLE
Duplicate log messages into syslog according to zowe.sysMessages

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
zowe.sysMessages is a new array that allows you to select messages that when found by the launcher will be duplicated into the system's log + zowe's own log, instead of just zowe's own log.

Requires: https://github.com/zowe/zowe-common-c/pull/391

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -162,21 +163,88 @@ struct {
...
+  Json *env;
+  int cfgGetStatus = cfgGetAnyC(configmgr, ZOWE_CONFIG_NAME, &env, 2, "zowe", "sysMessages");
+
+  if (cfgGetStatus != ZCFG_SUCCESS) { // No sysMessages found in Zowe configuration
+    return;
+  }
+  JsonArray *sys_messages = jsonAsArray(env);
 
+  int count = jsonArrayGetCount(sys_messages);
+  printf("\nWHATS OUR SYSMSG COUNT %d?", count);
```
Comment (timgerstel) [inline src/main.c:184]: Remove printf

File: src/main.c [hunk 2]
```diff
@@ -162,21 +163,88 @@ struct {
...
+  for (int i = 0; i < count; i++) {
+      char *sys_message = jsonArrayGetString(sys_messages, i);
+      printf("SYS MESSAGE HERE: %s\n", sys_message);
```
Comment (timgerstel) [inline src/main.c:187]: Remove printf

File: src/main.c [hunk 3]
```diff
@@ -1596,6 +1665,10 @@ int main(int argc, char **argv) {
   if (process_root_dir(configmgr)) {
     exit(EXIT_FAILURE);
   }
+  
+  /* TODO(?): sys_messages could be set earlier than this w/ known sysMessages w/o having
```
Comment (DivergentEuropeans) [inline src/main.c]: print start launcher message to syslog + print any exiting errors to syslog

Comment (DivergentEuropeans) [reply src/main.c]: done

**Review by 1000TurquoisePogs — APPROVED** [review]
lgtm

**Comment by DivergentEuropeans** [comment]
create Launcher's first changelog

File: src/main.c [hunk 4]
```diff
@@ -162,21 +163,79 @@ struct {
...
+      }
+  }
+  
+}
+
+#define INFO(fmt, ...)  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
+  printf("%s <%s:%d> %s INFO "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
+#define WARN(fmt, ...)  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
+  printf("%s <%s:%d> %s WARN "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)
+#define DEBUG(fmt, ...) check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
```
Comment (DivergentEuropeans) [inline src/main.c]: Maybe this part needs to be
`
if (zl_context.config.debug_mode) \
  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
  printf("%s <%s:%d> %s DEBUG "fmt, gettime().value, COMP_ID, zl_context.pid, zl_context.userid, ##__VA_ARGS__)`

**Pull request merged**